tests/reference/asr-array_01_decl-39cf894.json

```diff
@@ -6,7 +6,7 @@
     "outfile": null,
     "outfile_hash": null,
     "stdout": "asr-array_01_decl-39cf894.stdout",
-    "stdout_hash": "137a0c427925ba7da2e7151f2cf52bfa9a64fede11fe8d2653f20b64",
+    "stdout_hash": "2aa47467473392c970bb1ddde961e3007d4c157bb0ea507b5e0db4a4",
     "stderr": null,
```

Comment:
Just curious, why do we need these generated `.json` or `.stderr` files? Is there any place I could read on the importance of these additions? Thank you!

Reply:
These are part of the reference tests. Reference tests help us to ensure that the output being generated is as expected. The `.stderr` reference output is generated when there is an error being observed when using the compiler.  This could be a syntax/parser error, a semantic error or I think even a code generation error (this is probably rare, we mostly handle all errors at the semantic level). The `.stderr` reference tests help us ensure that there is indeed an error being observed for different cases where there is an error expected (For example, we expect an error to be throw when there is a type mismatch). The error message is usually descriptive and guides the user towards the correct approach (If it is not then that might be a bug and we need to improve it.)

Reply:
Hi, @Shaikh-Ubaid!
Got it. Thanks for the clear description.